framework/util/platform_debug_settings.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+** Copyright (c) 2022 Valve Corporation
+** Copyright (c) 2022 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include <cstdlib>
+
+#if defined(WIN32)
+
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+#include <windows.h>
+
+DWORD DisableDebugPopup(void)
+{
+    DWORD variable_length = GetEnvironmentVariableA("GFXRECON_NO_DEBUG_POPUP", nullptr, 0);
+
+    bool variable_exists = (variable_length > 0);
+
+    if (variable_exists)
+    {
+        _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
+    }
```

Comment:
Vague memory and the [**LunarG Suggested Code Development Workflow** document](https://docs.google.com/document/d/1W0W4Sm8hLC9Sh9xBwFJthPk7ZZzPxjqU68xDLZ6LyQY/edit#bookmark=id.4yvy0kvdgglg) suggest that other calls may also be of interest:
```
SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX);
_CrtSetReportMode(_CRT_ASSERT, _CRTDBG_MODE_FILE);
_CrtSetReportFile(_CRT_ASSERT, _CRTDBG_FILE_STDERR);
```
I'm actually not sure whether any of these are relevant or will work in this context; I'm just trying to be complete.  If investigating such is not something you have time for, I'll still happily take just disabling debug popups.

Reply:
In this case I've simply copied the code that is called for `--no-debug-popup`.  Those seem like good suggestions and we should come back to those.